src/gui/widgets/settings.cpp

```diff
@@ -585,14 +585,25 @@ const builder_window::window_resolution& get_window_builder(const std::string& t
 
 	VALIDATE(!resolutions.empty(), formatter() << "Window '" << type << "' has no resolutions.\n");
 
+	const builder_window::window_resolution* best_resolution = nullptr;
+	int best_resolution_score = 0;
+
 	for(const auto& res : resolutions) {
-		if(settings::screen_width <= res.window_width && settings::screen_height <= res.window_height) {
```

Comment:
> To clarify, in my opinion the window_width and window_height attributes should never be set less than 800 and 600 respectively


I i could assume that the wesnoth window size is always at lest 800x600 then you coudl ofrc course use 800x600 there, but i noticed that i can get a window size less than 800x600 by draggibng the window title to the upper left corner of the screen.